src/topology/homotopy/homotopy_group.lean

```diff
@@ -154,42 +314,140 @@ begin
       prop' := λ _ _ ⟨i,_⟩, i.elim0 }⟩]
 end
 
-/-- The 1-dimensional generalized loops based at `x` are in 1-1 correspondence with
-  paths from `x` to itself. -/
-@[simps] def gen_loop_one_equiv_path_self : gen_loop 1 x ≃ path x x :=
-{ to_fun := λ p, path.mk ⟨λ t, p (λ _, t), by {continuity, exact p.1.2}⟩
-    (p.boundary (λ _, 0) ⟨0, or.inl rfl⟩)
-    (p.boundary (λ _, 1) ⟨1, or.inr rfl⟩),
+/-- The 1-dimensional generalized loops based at `x` are in 1-1 correspondence with loops at `x`. -/
+@[simps] def gen_loop_one_equiv_loop : Ω^(fin 1) x ≃ Ω X x :=
```

Comment:
This should be a consequence of `gen_loop.loop_equiv` and `gen_loop_zero_equiv` (generalized to empty types) and functoriality of `Ω`

Reply:
Also generalize this to types with the `unique` condition.

Reply:
> This should be a consequence of `gen_loop.loop_equiv` and `gen_loop_zero_equiv` (generalized to empty types) and functoriality of `Ω`

I refrained from doing this suggested refactor because we don't have the functoriality yet; I added it as a TODO item. Generalization to `unique` is done.